x-pack/plugins/infra/public/components/log_stream/log_stream.tsx

```diff
@@ -208,6 +217,35 @@ Read more at https://github.com/elastic/kibana/blob/main/src/plugins/kibana_reac
     [isLoadingMore, fetchPreviousEntries, fetchNextEntries]
   );
 
+  const handleCloseFlyout = () => {
+    if (flyoutRef.current) {
+      flyoutRef.current.close();
+    }
+  };
+
+  const handleOpenLogFlyoutClick = useCallback(
+    (logEntryId) => {
+      const { Provider: KibanaReactContextProvider } = createKibanaReactContext({
+        http,
+        data,
+        uiSettings,
+        application,
+      });
+
+      flyoutRef.current = openFlyout(
+        <KibanaReactContextProvider>
```

Comment:
this is needed because `LogEntryFlyout` needs to call some services that exists in this context. `openFlyout` isolates the component into a separate mount which doesn't include it